agent/src/main/java/com/cloud/agent/Agent.java

```diff
@@ -150,6 +153,29 @@ public int value() {
     private String _keystoreSetupPath;
     private String _keystoreCertImportPath;
 
+    /**
+     * Virsh command to list the name of all VM instances starting with "i-".
+     */
+    protected static final String COMMAND_LIST_ALL_VMS = "virsh list --all --name | grep '^i-*'";
+
+    /**
+     * Virsh command to set the memory balloon stats period.<br><br>
+     * 1st parameter: the VM name;<br>
+     * 2nd parameter: the period (in seconds).
+     */
+    private static final String COMMAND_SET_MEM_BALLOON_STATS_PERIOD = "virsh dommemstat %s --period %s --live";
+
+    /**
+     * Virsh command to get the memory balloon tag from the VM's XML file.<br><br>
+     * 1st parameter: the VM name;<br>
+     */
+    private static final String COMMAND_GET_MEM_BALLOON_FROM_XML_FILE = "virsh dumpxml %s | grep '<memballoon model='";
```

Comment:
(comment for all virsh usages)
Is there a libvirt equivalent to do this? Forking to shell isn't a good idea with arbitrary strings `%s` that may be passed to it.

In case libvirt doesn't support that, you may also raise a MR with https://gitlab.com/libvirt/libvirt-java and then use those changes.

Reply:
@rohityadavcloud Unfortunately, there are no corresponding methods for these specific virsh commands.
I think it would be better if we solved the collection of memory stats for now (even if it is using virsh commands). At another time we could add into libvirt-java corresponding methods for these and other commands already used in ACS.